### DISCUSSION TITLE
How to use stringbuilder in transform

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5136] [Q&A] [answered]
I was trying to use the stringbuilder but not having much luck. How do I add strings to the stringbuilder and then read them out?

Tried this:
```
var strBuilder = new java.lang.StringBuilder();
   strBuilder.append('Hello");
   logger.info(strBuilder);
````

but get and error 

	The choice of Java method java.lang.StringBuilder.append matching JavaScript argument types (org.mozilla.javascript.Undefined) is ambiguous; candidate methods are: 
    class java.lang.AbstractStringBuilder append(java.lang.String)
    class java.lang.AbstractStringBuilder append(java.lang.Object)

**Comment by jonbartels** [comment, marked as answer]
Try this:
```
var strBuilder = new java.lang.StringBuilder();
   strBuilder.append('Hello");
   logger.info(strBuilder.toString());
```

Poorly worded explanation: The error is because the JS -> Java engine, Rhino, is trying to figure out what to do with `strBuilder` to cast it to a `string` type for `logger.info`. `strBuilder` has a `toString` method but it's also a `StringBuilder` object. Rhino is saying it can't figure out what to do, so you have to be explicit.
